mod/primitives/pkg/ssz/vector.go

```diff
@@ -73,6 +75,15 @@ func (l VectorBasic[B]) N() uint64 {
 func (l VectorBasic[B]) HashTreeRootWith(
 	merkleizer BasicMerkleizer[[32]byte, B],
 ) ([32]byte, error) {
+	var b B
+	// If this is a Vector of Bytes, we can use the optimized
+	// version of the merkleizer opposed to having to iterate through
+	// each byte by byte.
+	if _, ok := any(b).(Byte); ok {
+		return merkleizer.MerkleizeByteSlice(
```

Comment:
Rather than doing this special case why not just uniquely define the ByteVector type?